sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/Plan.java

```diff
@@ -44,4 +47,8 @@ default boolean isRecoveryPlan() {
     default boolean isDecommissionPlan() {
         return getName().equals(DECOMMISSION_PLAN_NAME);
     }
+
+    Optional<String> getNamespace();
```

Comment:
Instead of having the plans advertise the service they belong to, how about passing different customizer objects (each bond to a certain namespace) to each of the services when they're being built?

In other words, the logic here is something like this:
```
customization = new ExampleCustomization();
// common customization object, can't tell which service is invoking it:
service1 = new Service(namespace1, customization)
service2 = new Service(namespace2, customization)
```

when it could be:
```
customization = new ExampleCustomization();
// separate per-service subobjects, each bound to specific namespaces:
service1 = new Service(namespace1, customization.getForNamespace(namespace1))
service2 = new Service(namespace2, customization.getForNamespace(namespace2))
```

In the latter structure, the per-service subobjects would already know the context that they're operating in

Reply:
This might do it?: https://github.com/mesosphere/dcos-kubernetes/pull/540

Reply:
OK that sounds like a plan (get it?).